### PR TITLE
Implemented ctx struct in actions, do_path uses game-state structs now

### PR DESCRIPTION
**Pull request by yoonsjeong**
The following issues were completed in this PR: #191 #193 #250 #324 #329 
Here is a quick summary of each issue

Issue 191:
- altered do_path_action functions so that they use game-state functions (it now actually moves the player to the given room and path)

Issue 193:
- added tests to check for the added features in issue 191

Issue 250:
- refactored do_path, do_item, and do_item_item so that they either use the newly added ctx context struct, or do not use context at all
- refactored tests corresponding to each of these functions

Issue 324:
- simplified movement by removing the "ENTER" action_type (it will likely be added as a synonym later on)

Issue 329:
- a game-state fix in their game-state/tests/test-items.c that was causing travis CI to fail

### CODE REVIEW

File: src/action_management/src/actionmanagement.c [hunk 1]
```diff
@@ -91,36 +89,60 @@ int do_item_action(game_t *g, action_type_t *a, item_t *i, char **ret_string)
...
-    /* TODO: implement the rest of this function, using game state funcs
-     * Will perform the action if all checks pass (Sprint 4)
-     */
-    sprintf(string, "Requested action %s in direction %s into room %s",
-            a->c_name, p->direction, p->dest->room_id);
-    *ret_string = string;
-    return SUCCESS;
+    // validate existence of path and destination
+    if ((path_found == NULL) || (room_dest == NULL)) {
+        sprintf(string, "The path or room provided was invalid.");
```
[the remark this answers is not in the document]
Comment (yoonsjeong) [reply src/action_management/src/actionmanagement.c]: oops this was a typo -- fixed in newest commit!

File: src/action_management/tests/test_path_actions.c [hunk 2]
```diff
@@ -4,57 +4,75 @@
...
-{
-    int rc = execute_do_path_action("dummy", ITEM_ITEM);
+    /* FAIL TESTS */
+    check_do_path(ctx_test, action_invalid, path_north, room_origin, WRONG_KIND);
+    check_do_path(ctx_test, action_go, path_origin, room_origin, NOT_ALLOWED_PATH);
 
-    cr_assert_eq(rc, WRONG_KIND,
-                 "execute_do_item_item_action returned %d for wrong kind ITEM_ITEM expected WRONG_KIND (1)");
+    /* FREE VARIABLES */
+    chiventure_ctx_free(ctx_test);
```
[the remark this answers is not in the document]
Comment (yoonsjeong) [reply src/action_management/tests/test_path_actions.c]: thanks for catching this! I had mistakenly assumed ctx_free freed game resources too

**Comment by yoonsjeong** [comment]
@mpcsh I've fixed the Travis CI build failure! We should be ready to get this merged to master 👍

**Comment by yoonsjeong** [comment]
Closing statement:
The primary goal of this PR, to redo the do_path function and its tests is now part of master. In the process we also merged in the new context struct, which will further help with integration with the other teams.

Currently we are running sequel PRs #345 and #346 that will implement the do_item and do_item_item functions into actions/dev, and will create a PR akin to this one after those are approved.

**Comment by mpcsh** [comment]
Issue Score: ✓+

Comments:
Great work!

**Pull request merged**